src/hotspot/share/runtime/objectMonitor.hpp

```diff
@@ -139,20 +139,12 @@ class ObjectMonitor {
   // Enforced by the assert() in header_addr().
   volatile markWord _header;        // displaced object header word - mark
   WeakHandle _object;               // backward object pointer
-  typedef enum {
-    Free = 0,  // Free must be 0 for monitor to be free after memset(..,0,..).
-    New,
-    Old,
-    ChainMarker
-  } AllocationState;
-  AllocationState _allocation_state;
   // Separate _header and _owner on different cache lines since both can
-  // have busy multi-threaded access. _header, _object and _allocation_state
-  // are set at initial inflation. _object and _allocation_state don't
-  // change until deflation so _object and _allocation_state are good
-  // choices to share the cache line with _header.
+  // have busy multi-threaded access. _header and _object are set at initial
+  // inflation. The _object does not change, so it is a good choice to share
+  // its cache line with _header.
   DEFINE_PAD_MINUS_SIZE(0, OM_CACHE_LINE_SIZE, sizeof(volatile markWord) +
-                        sizeof(WeakHandle) + sizeof(AllocationState));
+                        sizeof(WeakHandle));
   // Used by async deflation as a marker in the _owner field:
```

Comment:
I have test with and without padding, I saw no difference.

Reply:
We've removed enough padding with this work already. If we
want to do more padding removal, then we need to use a
different RFE.

Reply:
Sure, this was more a FYI.

Reply:
Filed this new RFE:
    JDK-8256303 revisit ObjectMonitor padding between fields
    https://bugs.openjdk.java.net/browse/JDK-8256303